compiler/p/codegen/OMRCodeGenerator.cpp

```diff
@@ -502,24 +502,29 @@ OMR::Power::CodeGenerator::mulDecompositionCostIsJustified(
 
    // Notice: the total simple operations is (2*numOfOperations-1). The following checking is a heuristic
    //         on processors we still care about.
-   switch (self()->comp()->target().cpu.id())
+   switch (self()->comp()->target().cpu.getProcessorDescription().processor)
       {
-      case TR_PPCpwr630: // 2S+1M FXU out-of-order
+      case OMR_PROCESSOR_PPC_PWR630: // 2S+1M FXU out-of-order
+         TR_ASSERT_FATAL(self()->comp()->target().cpu.id() == TR_PPCpwr630, "TR_PPCpwr630");
          return (numOfOperations<=4);
 
-      case TR_PPCnstar:
-      case TR_PPCpulsar: // 1S+1M FXU in-order
+      case OMR_PROCESSOR_PPC_NSTAR:
+      case OMR_PROCESSOR_PPC_PULSAR: // 1S+1M FXU in-order
+         TR_ASSERT_FATAL(self()->comp()->target().cpu.id() == TR_PPCnstar || self()->comp()->target().cpu.id() == TR_PPCpulsar, "TR_PPCnstar, TR_PPCpulsar");
          return (numOfOperations<=8);
 
-      case TR_PPCgpul:
-      case TR_PPCgp:
-      case TR_PPCgr:    // 2 FXU out-of-order back-to-back 2 cycles. Mul is only 4 to 6 cycles
+      case OMR_PROCESSOR_PPC_GPUL:
+      case OMR_PROCESSOR_PPC_GP:
+      case OMR_PROCESSOR_PPC_GR:    // 2 FXU out-of-order back-to-back 2 cycles. Mul is only 4 to 6 cycles
+         TR_ASSERT_FATAL(self()->comp()->target().cpu.id() == TR_PPCgpul || self()->comp()->target().cpu.id() == TR_PPCgp || self()->comp()->target().cpu.id() == TR_PPCgr, "TR_PPCgpul, TR_PPCgp, TR_PPCgr");
          return (numOfOperations<=2);
 
-      case TR_PPCp6:    // Mul is on FPU for 17cycles blocking other operations
+      case OMR_PROCESSOR_PPC_P6:    // Mul is on FPU for 17cycles blocking other operations
+         TR_ASSERT_FATAL(self()->comp()->target().cpu.id() == TR_PPCp6, "TR_PPCp6");
          return (numOfOperations<=16);
 
-      case TR_PPCp7:    // Mul blocks other operations for up to 4 cycles
+      case OMR_PROCESSOR_PPC_P7:    // Mul blocks other operations for up to 4 cycles
+         TR_ASSERT_FATAL(self()->comp()->target().cpu.id() == TR_PPCp7, "TR_PPCp7");
```

Comment:
We are checking if the new API's are behaving correctly. For example, if the new processor version is OMR_PROCESSOR_PPC_PWR630, then old processor version must be TR_PPCpwr630. 
We have the more generic test somewhere else but they only get triggered by calling `is()` or `supportsFeature()`. Since we are doing a switch statement here, I just added more checks.